core/src/main/scala/kafka/log/Log.scala

```diff
@@ -444,6 +457,23 @@ class Log(@volatile var dir: File,
     }
   }
 
+  /*
+   * Increment the log start offset if the provided offset is larger.
+   * And delete old segments whose largest offset < log start offset.
+   */
+  def maybeIncrementLogStartOffsetAndDeleteRecords(offset: Long) {
```

Comment:
So, DeleteRecordsRequest can't be applied on a compacted topic? Should we return an error code in the response in that case?